test/tests/functional/pbs_mom_dynamic_resource.py

```diff
@@ -373,28 +377,33 @@ def test_mom_dyn_res_permissions(self):
         self.mom.add_config({'foo': mom_config_str})
 
         # give write permission to group and others
-        self.du.chmod(path=fp, mode=0766)
+        self.du.chmod(path=fp, mode=0766, sudo=True)
         self.check_access_log(fp)
 
         # give write permission to group
-        self.du.chmod(path=fp, mode=0764)
+        self.du.chmod(path=fp, mode=0764, sudo=True)
         self.check_access_log(fp)
 
         # give write permission to others
-        self.du.chmod(path=fp, mode=0746)
+        self.du.chmod(path=fp, mode=0746, sudo=True)
         self.check_access_log(fp)
 
         # give write permission to user only
-        self.du.chmod(path=fp, mode=0744)
-        self.check_access_log(fp, exist=False)
+        self.du.chmod(path=fp, mode=0744, sudo=True)
+        root_priv = self.du.run_cmd(
```

Comment:
if you just want to check that the directory exists then why not do os.path.isdir()?

Reply:
I'm trying to check if the user running the test is root or not. I'll change it to do os.getuid() instead